pennylane_qiskit/qiskit_device.py

```diff
@@ -315,7 +318,8 @@ def rotate_basis(self, obs, wires, par):
     def pre_measure(self):
         for e in self.obs_queue:
             # Add unitaries if a different expectation value is given
-            if hasattr(e, "return_type") and e.return_type == Sample:
+            # Exclude unitary_simulator as it does not support memory=True
+            if hasattr(e, "return_type") and e.return_type == Sample and self.backend_name != 'unitary_simulator':
```

Comment:
"Backend support for memory=True now checked when that kwarg is passed. QiskitError results if not supported." - [from the release notes of version 0.14.0](https://qiskit.org/documentation/release_notes.html)